rfc/rfc-0014-binding-injection.md

```diff
@@ -0,0 +1,131 @@
+# RFC-0014: Binding Injection
+
+**Authors:** @scothis
+
+**Status:**
+
+**Pull Request URL:** [#1382](https://github.com/projectriff/riff/pull/1382)
+
+**Superseded by:** N/A
+
+**Supersedes:** N/A
+
+**Related:** [RFC 0002 Bindings](./rfc-0002-bindings.md), [RFC 0008 Pod Templates](./rfc-0008-pod-templates.md)
+
+
+## Problem
+
+[RFC 0002 Bindings](./rfc-0002-bindings.md) defines a interface for a resource to expose metadata and secrets to be consumed by another resource. It leverages the [CNB Binding](https://github.com/buildpack/spec/blob/master/extensions/bindings.md) spec for how the service metadata and secrets are exposed inside a container. While a binding can be consumed manually, it's difficult to manage as it requires defining two volumes, and two volume mounts and a environment variable per container.
+
+### Anti-Goals
+
+This RFC has no opinion as to the type of binding being injected, or the consumer of the binding.
+
+## Solution
+
+Knative Bindings (yea, the term is overloaded) define an abstract way for reconcilers to operate on resources by duck typing known elements of the resource without requiring pre-knowledge or full type knowledge at runtime. By leveraging resources that use a PodTemplateSpec, it's possible to inject volumes, volume mounts and environment variables into an existing resource (even if owned and controlled by another resource). With [RFC 0008 Pod Templates](./rfc-0008-pod-templates.md) the riff Deployers and Processor resources implement the PodSpecable duck type.
+
+A duck type for exposing a binding is defined in RFC 0002. While not as wide spread, the Stream resource currently implements this duck type. Combining the Binding and PodSpecable duck types allows for a generic means to inject binding metadata and secrets into a consuming workload. A new CRD can express this injection:
```

Comment:
i suggest qualifying the word "Binding" here so that readers don't confuse the CNB service binding idea with the more generic Knative Binding. I also suggest not using the word "Binding" to qualify the kind of duck type for the same reason.

Reply:
Thanks, that's better now - resolving.